lib/screens/v2/candidate_generator/widgets/elevator_closures.ex

```diff
@@ -22,12 +46,24 @@ defmodule Screens.V2.CandidateGenerator.Widgets.ElevatorClosures do
     with {:ok, location_context} <- fetch_location_context_fn.(PreFare, parent_station_id, now),
          {:ok, parent_station_map} <- Stop.fetch_parent_station_name_map(),
          {:ok, alerts} <- fetch_elevator_alerts_with_facilities_fn.() do
-      elevator_closures = relevant_alerts(alerts)
-      icon_map = get_icon_map(elevator_closures, parent_station_id)
+      elevator_alerts = relevant_alerts(alerts)
+
+      icon_map = get_icon_map(elevator_alerts, parent_station_id)
+
+      elevator_closures =
+        elevator_alerts
+        |> Enum.flat_map(&elevator_closure/1)
+
+      elevator_closure_ids =
+        elevator_closures
+        |> Enum.filter(&relevant_closure?(&1, elevator_closures))
+        |> Enum.map(fn %Closure{alert_id: alert_id} -> alert_id end)
 
       [
         %ElevatorStatusWidget{
-          alerts: elevator_closures,
+          alerts:
+            elevator_alerts
+            |> Enum.filter(fn %Alert{id: id} -> id in elevator_closure_ids end),
```

Comment:
`elevator_closure_ids` is a list here, requiring a linear-time traversal for each item in `elevator_alerts`, meaning this takes N² time overall. Admittedly this couldn't possibly matter with the values of N we'd expect here, but in this kind of situation I'd consider it good practice to create a set (`|> MapSet.new()`).

Reply:
great callout!